### PR TITLE
Jenkins separate deploy and sign builds

### PR DESCRIPTION
**Pull request by joker234**
# Changes proposed in this pull request:
## Type of change
- CI Update

## Description
- Signed commits
- Separate snapshot and release deploy stages

## Corresponding issue
None

## New or changed dependencies:
None

# Checklist
No java code edited, so none of the following applies:
- [ ] ~~My code follows the [code-style](https://github.com/GIScience/oshdb/blob/master/CONTRIBUTING.md) rules and I have checked on the [static analyses](https://jenkins.ohsome.org/job/oshdb/view/change-requests/) and [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) (if applicable) results~~
- [ ] ~~I have commented my code~~
- [ ] ~~I have written javadoc (required for public methods)~~
- [ ] ~~I have added sufficient unit tests~~
- [ ] ~~I have made corresponding changes to the [documentation](https://github.com/GIScience/oshdb/tree/master/documentation)~~
- [ ] ~~I have updated the [CHANGELOG.md](https://github.com/GIScience/oshdb/blob/master/CHANGELOG.md)~~
- [ ] ~~I have adjusted the [examples](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-examples) and [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) if necessary~~

### CODE REVIEW

**Comment by joker234** [comment]
Hint: this issue doesn't include the automatic deployment to Central Maven. This will be a followup to this issue and depends on these changes.

File: Jenkinsfile [hunk 1]
```diff
@@ -38,10 +45,32 @@ pipeline {
...
+        failure {
+          rocketSend channel: 'jenkinsohsome', message: "Deployment of oshdb-build nr. ${env.BUILD_NUMBER} *failed* on Branch - ${env.BRANCH_NAME}  (<${env.BUILD_URL}|Open Build in Jenkins>). Latest commit from  ${author}. Is Artifactory running?" , rawMessage: true
+        }
+      }
+    }
+
+    stage ('Deploy Release') {
+      when {
+        allOf {
+          tag pattern: "(^[0-9]+\$)|(^(([0-9]+)(\\.))+([0-9]+)?\$)", comparator: "REGEXP"
```
Comment (tyrasd) [inline Jenkinsfile]: Maybe we could tweak the regex a bit, to allow beta/RC releases? And I believe we would never do a major-version only release like `7`, so the first part of the regex could be dropped.

From what I can [see](https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN400), maven uses versions in the style of `<major version>.<minor version>.<incremental version>-<qualifier>`, where all parts are numbers, except for the qualifier which can be a string and the SNAPSHOT qualifier has some special meaning… I guess if we could limit ourselves to a fixed set of qualifiers (`alpha-[0-9]+`, `beta-[0-9]+`, `RC[0-9]+`).

Comment (joker234) [reply Jenkinsfile]: How about: `/^([0-9]+(\.[0-9]+)*?)(-(RC|beta-|alpha-)[0-9]+)?$/` ([regex101](https://regex101.com/r/7pFHIE/1))

Or better: `/^([0-9]+(\.[0-9]+)*)(-(RC|beta|alpha)-?[0-9]+)?$/` ([regex101](https://regex101.com/r/XJVDCz/1))

Comment (tyrasd) [reply Jenkinsfile]: let's use the first one

Comment (joker234) [reply Jenkinsfile]: right, so I use `RELEASE_REGEX = /^([0-9]+(\.[0-9]+)*)(-(RC|beta-|alpha-)[0-9]+)?$/`

Comment (joker234) [reply Jenkinsfile]: right, so I use `RELEASE_REGEX = /^([0-9]+(\.[0-9]+)*)(-(RC|beta-|alpha-)[0-9]+)?$/` for both, tag name and version number

**Pull request merged**